patches/tModLoader/Terraria/Player.TML.cs

```diff
@@ -139,5 +140,61 @@ public partial class Player
 			bool four = modBiomeFlags.Cast<bool>().Contains(true);
 			return !(one || two || three || four);
 		}
+
+		/// <summary>
+		/// Invoked at the end of loading vanilla player data from files to fix stuff that isn't initialized coming out of load.
+		/// Corrects the player.lavaMax time, wingsLogic, and no fall dmg to be accurate for the provided items in accessory slots.
+		/// </summary>
+		internal static void LoadPlayerLastMinuteFixes(Item item, Player newPlayer) {
```

Comment:
Not a vanilla bug per se - the method really only runs during player select screen, so it doesn't really have any ability to know the slot is or isn't available.
I left it as private primarily because this method is more decorative than practical - it doesn't actually change anything in terms of rendering on player select nor functionality once in world (beyond the first tick).

I'll change it to public with the additional notes, but don't think we need to think too much about this method beyond that it has been matched for purposes of vanilla consistency.